### PR TITLE
Add javadoc for SqlSelectFlow's methods - part two

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Issue related to this PR: [84](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/84)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 1]
```diff
@@ -295,17 +292,15 @@ private static URI getQueuedUri(QueryId queryId, Slug slug, long token, UriInfo
...
+     * Instantiates an object that contains all information about query metadata.
      *
-     * @return a {@link QueryResults} object.
+     * @param queryId an object containing the query identifier
+     * @param nextUri an object that contains the URI where information about query can be retrieved
+     * @param queryError an object responsible for reporting the error that happened during the query's execution
+     * @param uriInfo an object that will be used to construct the query representation in web interface
+     * @param elapsedTime the time elapsed until the query obtains its response
+     * @param queuedTime the time the query waited to be executed
+     * @return an object with all query's metadata
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:303]: why not keep the link to {@link QueryResults} in the return statement?

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 2]
```diff
@@ -441,6 +436,15 @@ public void destroy()
             sessionContext.getIdentity().destroy();
         }
 
+        /**
+         * Instantiates an object that contains all information about query metadata.
+         *
+         * @param token a generated token that identifies the client's request used to obtains the response from server
+         * @param uriInfo an object used to obtains the URI where query can be retrieved in server or construct the query representation inside
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:443]:
```suggestion
         * @param uriInfo an object used to obtain the URI where the query can be retrieved from server or construct the query representation inside
```

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 3]
```diff
@@ -441,6 +436,15 @@ public void destroy()
...
+         * the web interface
+         * @param dispatchInfo an object responsible for taking the coordinator's location, response time and waiting time
+         * @return an object with all query's metadata
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:446]: could reference a {@link QueryResults} in the return statement

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 4]
```diff
@@ -458,18 +462,12 @@ private QueryResults createQueryResults(long token, UriInfo uriInfo, DispatchInf
...
-         *
-         * @param token generated token
-         * @param uriInfo an object containing the uri metadata.
-         * @param dispatchInfo an object responsible for taking the coordinator's location, response time
-         * and waiting time
+         * Gets the URI where information about query can be found inside server if it is not completed.
          *
-         * @return a {@link URI} object
+         * @param token a generated token that identifies the client's request used to obtains the response from server
+         * @param uriInfo an object used to obtains the URI where query can be retrieved in server
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:468]:
```suggestion
         * @param uriInfo an object used to obtain the URI where the query can be retrieved from the server
```

File: presto-main/src/main/java/io/prestosql/server/protocol/Slug.java [hunk 5]
```diff
@@ -52,13 +52,11 @@ private Slug(byte[] slugKey)
...
     /**
-     * Computes a hash code based on the data that have been provided
-     * and turns it into a string slug.
+     * Computes a hash code based on data about the query state and the received token.
      *
-     * @param context fixed set of constants
-     * @param token generated token
-     *
-     * @return a slug as a string.
+     * @param context a constant that represents if query is running or is queued
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/protocol/Slug.java:57]:
```suggestion
     * @param context a constant that represents if query is running or queued
```

File: presto-main/src/main/java/io/prestosql/server/protocol/Slug.java [hunk 6]
```diff
@@ -52,13 +52,11 @@ private Slug(byte[] slugKey)
...
+     * @param token a generated token that identifies the client's request used to obtains the response from server
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/protocol/Slug.java:58]:
```suggestion
     * @param token a generated token that identifies the client's request used to obtain the response from server
```

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java (hunk 3 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:446]: https://softwareengineering.stackexchange.com/questions/199293/is-repeating-links-to-the-same-class-in-a-single-javadoc-comment-a-bad-practice

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java (hunk 1 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:303]: https://softwareengineering.stackexchange.com/questions/199293/is-repeating-links-to-the-same-class-in-a-single-javadoc-comment-a-bad-practice

Comment (felipe-dna) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:303]: The `params` and `return` descriptions must to be aligned.

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 7]
```diff
@@ -441,6 +436,15 @@ public void destroy()
...
         }
 
+        /**
+         * Instantiates an object that contains all information about query metadata.
+         *
+         * @param token a generated token that identifies the client's request used to obtains the response from server
+         * @param uriInfo an object used to obtain the URI where the query can be retrieved from server or construct the query representation inside
+         * the web interface
+         * @param dispatchInfo an object responsible for taking the coordinator's location, response time and waiting time
+         * @return an object with all query's metadata
```
Comment (felipe-dna) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:446]: The `params` and `return` descriptions must to be aligned.

File: presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java [hunk 8]
```diff
@@ -458,18 +462,12 @@ private QueryResults createQueryResults(long token, UriInfo uriInfo, DispatchInf
...
-         * @param uriInfo an object containing the uri metadata.
-         * @param dispatchInfo an object responsible for taking the coordinator's location, response time
-         * and waiting time
+         * Gets the URI where information about query can be found inside server if it is not completed.
          *
-         * @return a {@link URI} object
+         * @param token a generated token that identifies the client's request used to obtains the response from server
+         * @param uriInfo an object used to obtain the URI where the query can be retrieved from the server
+         * @param dispatchInfo an object responsible for taking the coordinator's location, response time and waiting time
+         * @return the URI where information about query can be found inside server if it is not completed.
```
Comment (felipe-dna) [inline presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java:470]: The `params` and `return` descriptions must to be aligned.

File: presto-main/src/main/java/io/prestosql/server/protocol/Slug.java [hunk 9]
```diff
@@ -52,13 +52,11 @@ private Slug(byte[] slugKey)
...
-     * and turns it into a string slug.
+     * Computes a hash code based on data about the query state and the received token.
      *
-     * @param context fixed set of constants
-     * @param token generated token
-     *
-     * @return a slug as a string.
+     * @param context a constant that represents if query is running or is queued
+     * @param token a generated token that identifies the client's request used to obtain the response from server
+     * @return a hash code based on data about the query state and the received token
```
Comment (felipe-dna) [inline presto-main/src/main/java/io/prestosql/server/protocol/Slug.java:59]: The `params` and `return` descriptions must to be aligned.

File: presto-main/src/main/java/io/prestosql/server/protocol/Slug.java [hunk 10]
```diff
@@ -70,6 +68,14 @@ public String makeSlug(Context context, long token)
                 .toString();
     }
 
+    /**
+     * Checks if a slug is valid based on the received context and token.
+     *
+     * @param context a constant that represents if query is running or is queued
+     * @param slug a hash code based on data about the query state and the received token
+     * @param token a generated token that identifies the client's request used to obtain the response from server
+     * @return a flag that indicates if received slug is valid for the respective context and token
```
Comment (felipe-dna) [inline presto-main/src/main/java/io/prestosql/server/protocol/Slug.java:77]: The `params` and `return` descriptions must to be aligned.
